eng/native/configurecompiler.cmake

```diff
@@ -218,6 +218,14 @@ elseif (CLR_CMAKE_HOST_ARCH_ARM64)
   set(ARCH_HOST_NAME arm64)
   add_definitions(-DHOST_ARM64)
   add_definitions(-DHOST_64BIT)
+elseif (CLR_CMAKE_TARGET_ARCH_WASM)
+  set(ARCH_HOST_NAME wasm)
+  add_definitions(-DHOST_WASM=1)
+  add_definitions(-DFEATURE_64BIT_ALIGNMENT=1)
+  # The -fms-extensions enable the stuff like __if_exists, __declspec(uuid()), etc.
+  add_compile_options(-fms-extensions)
+  add_compile_options(-Wno-invalid-offsetof)
+  add_compile_options(-Wno-tautological-undefined-compare) # this == NULL warning suppression
```

Comment:
`ms-extensions` I believe is preventing 
```
In file included from E:\GitHub\runtimelab\src\coreclr\src\nativeaot\Runtime\allocheap.cpp:9:
         E:/GitHub/runtimelab/src/coreclr/src/nativeaot/Runtime/./PalRedhawk.h:658:1: error: unknown type name '__forceinline'
         __forceinline
         ^
         In file included from E:\GitHub\runtimelab\src\coreclr\src\nativeaot\Runtime\allocheap.cpp:9:
         In file included from E:/GitHub/runtimelab/src/coreclr/src/nativeaot/Runtime/./PalRedhawk.h:873:
```
`-Wno-invalid-offsetof` is stopping:
```
        In file included from E:\GitHub\runtimelab\src\coreclr\src\nativeaot\Runtime\allocheap.cpp:21:
         E:/GitHub/runtimelab/src/coreclr/src/nativeaot/Runtime/./slist.inl:41:54: warning: offset of on non-standard-layout type 'AllocHeap::BlockListElem' [-Winvalid-offsetof]
             return dac_cast<PTR_PTR_T>(dac_cast<TADDR>(pT) + offsetof(T, m_pNext));
                                                              ^           ~~~~~~~
         E:\GitHub\emsdk\upstream\emscripten\system\include\libc\stddef.h:20:32: note: expanded from macro 'offsetof'
         #define offsetof(type, member) __builtin_offsetof(type, member)
                                        ^                        ~~~~~~
         E:/GitHub/runtimelab/src/coreclr/src/nativeaot/Runtime/./slist.inl:245:14: note: in instantiation of member function 'DefaultSListTraits<AllocHeap::BlockListElem, DoNothingFailFastPolicy>::GetNextPtr' requested here
             *Traits::GetNextPtr(pItem) = *m_ppCur;
                      ^
         E:/GitHub/runtimelab/src/coreclr/src/nativeaot/Runtime/./slist.inl:85:13: note: in instantiation of member function 'SList<AllocHeap::BlockListElem, DefaultSListTraits<AllocHeap::BlockListElem, DoNothingFailFastPolicy>>::Iterator::Insert' requested here
             Begin().Insert(pItem);
                     ^
         E:\GitHub\runtimelab\src\coreclr\src\nativeaot\Runtime\allocheap.cpp:100:17: note: in instantiation of member function 'SList<AllocHeap::BlockListElem, DefaultSListTraits<AllocHeap::BlockListElem, DoNothingFailFastPolicy>>::PushHead' requested here
             m_blockList.PushHead(pBlock);
                         ^
         1 warning generated.
```
`-Wno-tautological-undefined-compare` is for:
```
         E:\GitHub\runtimelab\src\coreclr\src\nativeaot\Runtime\eetype.cpp:24:9: warning: 'this' pointer cannot be null in well-defined C++ code; comparison may be assumed to always evaluate to false [-Wtautological-undefined-compare]
             if (this == NULL)
                 ^~~~    ~~~~
         1 warning generated.
```

Reply:
I see. These and warning options are in this file under `CLR_CMAKE_HOST_UNIX`. Would it make sense to change the condition on that block to !MSVC or something similar instead of duplicating them here?

Reply:
Thanks, move as suggested to not duplicate.